libcob/jor.c

```diff
@@ -0,0 +1,464 @@
+/*
+   Copyright (C) 2023 Free Software Foundation, Inc.
+   Written by Emilien Lemaire and Fabrice Le Fessant.
+
+   This file is part of GnuCOBOL.
+
+   The GnuCOBOL compiler is free software: you can redistribute it
+   and/or modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of the
+   License, or (at your option) any later version.
+
+   GnuCOBOL is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with GnuCOBOL.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <stdarg.h>
+
+#include "config.h"
+
+/* include internal and external libcob definitions, forcing exports */
+#define COB_LIB_EXPIMP
+#include "coblocal.h"
+
+#include "tarstamp.h"
+#include "common.h"
+
+#include <sys/types.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#ifdef	_WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#undef MOUSE_MOVED
+#include <process.h>
+#include <io.h>
+#include <fcntl.h>	/* for _O_BINARY only */
+#endif
+
+#include <sys/time.h>
+
+/* Remember static and dynamic configuration */
+static cob_global               *cobglobptr = NULL;
+static cob_settings             *cobsetptr = NULL;
+static const char* cob_jor_filename ;
+
+static int is_active = 0;
+static char* jor_buffer ;
+static int jor_size ;
+static char* jor_position ;
+static int jor_name_counter ;
+
+static struct timeval tv0;
+
+static int field_secs ;
+static int field_usecs ;
+static int field_status ;
+static int field_reason ;
+static int field_file ;
+static int field_name ;
+static int field_truncated ;
+static int field_start ;
+static int field_exit ;
+static int field_duration ;
+static int field_time ;
+static int field_args ;
```

Comment:
likely `unsigned int`, if you don't need a negative marker (and if that is used, it is often useful to double-check if that's a  good implementation)